### ISSUE TITLE
Inconsistent behavior with std::map/unordered_map when using piecewise constructor in "emplace"

### ISSUE THREAD

**Issue by chineshboy** [greg7mdp/parallel-hashmap#104]
I am trying to use uncopyable & unmovable types (precisely atomic types) as the values in flat_hash_map/parallel_flat_hash_map. With std::map/unordered_map I was able to call the emplace function as:
```
std::unordered_map<long, std::atomic_long> m;
m.emplace(std::piecewise_construct,
          std::forward_as_tuple(round),
          std::forward_as_tuple(1));
```
, but with phmap::flat_hash_map/parallel_flat_hash_map I got compile errors:
```
phmap::flat_hash_map<long, std::atomic_long> m2;
m2.emplace(std::piecewise_construct,
           std::forward_as_tuple(round),
           std::forward_as_tuple(1));
```

> /usr/local/Cellar/gcc/10.2.0_1/include/c++/10.2.0/ext/new_allocator.h:150:4: error: use of deleted function 'constexpr std::pair<_T1, _T2>::pair(std::pair<_T1, _T2>&&) [with _T1 = long int; _T2 = std::atomic<long int>]'
  150 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }

**Comment by michaeleisel** [comment]
I'm having this issue as well

**Comment by greg7mdp** [comment]
I'll have a look but not probably for a couple days, as I'm overbooked right now.

**Comment by michaeleisel** [comment]
To be fair, if this is a flat hash map, then it will at least need one of copying or moving.

**Comment by greg7mdp** [comment]
Yes, @michaeleisel is exactly right, the `flat_*` hash maps store the entries directly in a memory array, so either move or copy is needed when the array is resized.
However, for non-copyable or movable items, you can use the `node_*` hash maps.
